examples/gno.land/r/demo/todolist/todolist.gno

```diff
@@ -0,0 +1,179 @@
+package todolist
+
+import (
+	"bytes"
+	"strconv"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/todolist"
+	"gno.land/p/demo/ufmt"
+)
+
+// State variables
+var (
+	todolistTree *avl.Tree
+	tlid         int
+)
+
+// Constructor
+func init() {
+	todolistTree = avl.NewTree()
+}
+
+func NewTodolist(title string) (int, string) {
+	// Create new Todolist
+	tl := todolist.NewTodoList(title)
+	// Update AVL tree with new state
+	todolistTree.Set(strconv.Itoa(tlid), tl)
+	tlid++
+	return tlid - 1, "created successfully"
+}
+
+func AddTask(todolistID int, title string) string {
+	// Get Todolist from AVL tree
+	tl, ok := todolistTree.Get(strconv.Itoa(todolistID))
+	if !ok {
+		return "Todolist not found"
+	}
+
+	// get the number of tasks in the todolist
+	id := tl.(*todolist.TodoList).Tasks.Size()
+
+	// create the task
+	task := todolist.NewTask(title)
+
+	// Cast raw data from tree into Todolist struct
+	tl.(*todolist.TodoList).AddTask(id, task)
+
+	return "task added successfully"
+}
+
+func ToggleTaskStatus(todolistID int, taskID int) string {
+	// Get Todolist from AVL tree
+	tl, ok := todolistTree.Get(strconv.Itoa(todolistID))
+	if !ok {
+		return "Todolist not found"
```

Comment:
Please consider swapping these returns for `panic`s, these methods should only return a `string` if they succeed